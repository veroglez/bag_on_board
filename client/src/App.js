import React, { useEffect } from 'react';
import { useStateValue } from 'common/context/contextHooks';
import C from 'common/constants';
import fetch from 'common/fetch';
import { Header } from './components';

const { ENDPOINT } = C;

const onFetch = async ({ dispatch }) => {
  const service = 'cards';
  const cards = await fetch({ ENDPOINT, service }).catch(err => err);
  dispatch({
    type: 'update',
    payload: {
      cards,
    },
  });
};

const App = () => {
  const [state, dispatch] = useStateValue();
  const { l10n } = state;

  useEffect(() => { onFetch({ dispatch }); }, []);
 console.log(state)
  return (
    <div className="App">
      <Header />
    </div>
  );
};

export default App;
