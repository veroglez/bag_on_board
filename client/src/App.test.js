
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Context from 'common/context/contextHooks';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });
global.fetch = require('jest-fetch-mock');

const STATE = {
  l10n: { ADD: 'add' },
  cards: [
    { name: 'Vero', bags: '3' },
  ],
};

const HOC = () => (
  <Context.StateContext.Provider value={[STATE]}>
    <App />
  </Context.StateContext.Provider>
);

const flushPromises = async () => new Promise(resolve => setImmediate(resolve));
const dispatch = jest.fn();

describe('<App>', () => {
  it('render', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispath is called with', async () => {
    fetch.mockResponseOnce(JSON.stringify(STATE.characters));

    jest
      .spyOn(Context, 'useStateValue')
      .mockImplementation(() => [STATE, dispatch]);
    mount(<HOC />);

    await flushPromises();
    expect(dispatch).toHaveBeenCalled();
  });
});
