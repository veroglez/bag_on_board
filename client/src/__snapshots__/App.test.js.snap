// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> render 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      <span>
        add
      </span>
    </button>
  </div>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <p
        className="name"
      >
        Vero
      </p>
      <p
        className="bags"
      >
        3
      </p>
    </div>
  </div>
</div>
`;
